hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/SimpleKeyProvider.java

```diff
@@ -49,6 +51,27 @@ public String getStorageAccountKey(String accountName, Configuration rawConfig)
       LOG.warn("Unable to get key from credential providers. {}", ioe);
     }
 
+    // Validating the key.
+    try {
+      validateStorageAccountKey(key);
+    } catch (InvalidConfigurationValueException e) {
+      e.printStackTrace();
```

Comment:
Throw it; we want to fail fast rather than wait for auth failures talking to the far end.

Reply:
Will it be fine to surround the ```throw new InvalidConfigurationValueException()``` with one more try-catch block? Since adding an exception to the method would break a test already present.

Reply:
Just throw KeyProviderException like it is done in line 49. Or better to call validateStorageAccountKey() just after L48 only so that you don't have to try and catch separately.

Reply:
I tried to place at L48 as well in the beginning but a test was getting broken, I found at that it expects a certain exception, it passes a null key, which makes the validate() method throw a different error. I think by adding a condition in the catch to throw that particular error if the key is null might solve this. So, I'll be changing it in the next commit.

Reply:
don't worry about breaking the test, we've moved the vaildation and should expect it. Instead the test will need to handle the way invalid configurations are raised

Reply:
Yes, I think the test is fine, I just needed to throw this specific exception when the config value is null which this test is trying to achieve, so I'll throw the exception when it is null else, it'll be like it always has been.